Mage/src/main/java/mage/cards/CardImpl.java

```diff
@@ -614,12 +614,17 @@ public boolean isTransformable() {
     @Override
     public final Card getSecondCardFace() {
         // init card side on first call
-        if (secondSideCardClazz == null && secondSideCard == null) {
+        if (secondSideCardClazz == null) {
             return null;
         }
 
         if (secondSideCard == null) {
             secondSideCard = initSecondSideCard(secondSideCardClazz);
+            if (secondSideCard != null && secondSideCard.getSpellAbility() != null) {
+                secondSideCard.getSpellAbility().setSourceId(this.getId());
+                secondSideCard.getSpellAbility().setSpellAbilityType(SpellAbilityType.BASE_ALTERNATE);
+                secondSideCard.getSpellAbility().setSpellAbilityCastMode(SpellAbilityCastMode.TRANSFORMED);
+            }
```

Comment:
Miss same changes in `getMeldsToCard`?

![shot_230413_180946](https://user-images.githubusercontent.com/8344157/231785586-0e980e03-7134-420e-af56-a792917f8b08.png)

Reply:
I don't think this is necessary as you'll never cast a card melded